### DISCUSSION TITLE
How to install and link wakatime on your github profile

### DISCUSSION THREAD

**Discussion by MECR-0209** [community/community#116451] [Programming Help]
## Before starting to work on GitHub, follow these steps:

- Access the WakaTime website and register at https://wakatime.com/.
- After registering, get the WakaTime API key.
- Integrate WakaTime with your development environment, like VS Code. To do this, follow these instructions:
  - In VS Code, go to extensions and install WakaTime.
  - After installation follow the instructions on the page, write something in one of your files and restart VS Code.
  - Go to the WakaTime website and it will already show that it is linked, if nothing appears just wait a little.
- In your WakaTime account, you will find the API key that will be used later.

After following these steps, you'll be ready to use WakaTime integrated into your development flow on GitHub.

### Just go to settings
![image](https://github.com/community/community/assets/137231429/c9e7e4a7-d298-4adc-a1cd-499621bb5cbf)

### Click on account and there is your key
![image](https://github.com/community/community/assets/137231429/bfc2fb34-4075-4b9d-9a2f-cde2ca2386b3)

# Steps to link hours to your profile

# 1- First create a GitHub key

### Go to settings
![image](https://github.com/community/community/assets/137231429/c1e3ff0a-f3d0-40a4-b45d-e726c92813b2)

### Click on the last option
![image](https://github.com/community/community/assets/137231429/64b3723b-1049-46dd-bb3d-40eacd660971)

![image](https://github.com/community/community/assets/137231429/37d7aaab-effb-44ac-9203-aa778386b90c)

![image](https://github.com/community/community/assets/137231429/e1ceba43-0516-4fd4-ac02-b64b3e270ba9)

### I saw in some places that they put permission on only some of these options, but I decided to allow everything on mine so I can't really say exactly which permission you need.

![image](https://github.com/community/community/assets/137231429/54d2798b-2150-4f02-8ce0-41537a4e20cd)

### In your repository README go to settings
![image](https://github.com/community/community/assets/137231429/f2fce60b-1347-4a5c-96af-b336831110fe)

### Then open your secrets
![image](https://github.com/community/community/assets/137231429/c5759bf3-04fb-4784-b9c8-0c0f848e1d04)

### There you place your GitHub key that you just copied and the WakaTime key
![image](https://github.com/community/community/assets/137231429/d7bb08be-b743-474f-bca3-002f0dd8a51d)

## 3- In your README file, copy and paste this into your code:

```
 <!--START_SECTION:waka-->
<!--END_SECTION:waka-->
```

### just create a new file

![image](https://github.com/community/community/assets/137231429/0b6e13c0-16f7-4a3e-8b8c-40a35da677da)

### and then just type the following: /.github/workflows/whatever_you_want_to_name.yml

![image](https://github.com/community/community/assets/137231429/903284ab-3fad-44d9-acae-d823823990ac)

### You just created a workflow and a yml file where you will put this code:

```
name: Work Stats Readme

on:
  workflow_dispatch:
  schedule:
    # Runs every 2 hours
    - cron: "0 */2 * * *"

jobs:
  update-readme:
    name: Update this repo's README
    runs-on: ubuntu-latest
    steps:
      - uses: athul/waka-readme@master 
        with:
          WAKATIME_API_KEY: ${{ secrets.WAKATIME_API_KEY }}
```

## Go to the settings of your README repository, click on ACTIONS and GENERAL.
![image](https://github.com/community/community/assets/137231429/0ea5308e-f9c9-41a4-b7c1-ad7497fa8259)

### A little further down on the page there are these options, leave it as it is in the image and click on SAVE
![image](https://github.com/community/community/assets/137231429/bd11d3c0-dd4a-4657-b73b-e9612876565f)

# Now try to run your workflow.

## In your repository, go to ACTIONS.
![image](https://github.com/community/community/assets/137231429/fcd82ae8-9e14-44b3-a47f-39ea881bca21)

## Then, click on your workflow and click run.
![image](https://github.com/community/community/assets/137231429/714c8b8a-0004-442f-8eee-ac28639ce500)
![image](https://github.com/community/community/assets/137231429/18ef1e6c-3515-4db3-9c71-c2709ef00d01)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Mohammedbbk** [comment]
Nice explanation!

**Comment by sharathsDev** [comment]
![image](https://github.com/user-attachments/assets/3e6c9ef5-9657-41b8-8a1f-b1266cc2e69f)

**Reply by sharathsDev** [reply]
I fond solution, copy paste below in your yml.
name: Waka Readme

on:
  schedule:
    # Runs at 12am IST
    - cron: '30 18 * * *'
  workflow_dispatch:
jobs:
  update-readme:
    name: Update Readme with Metrics
    runs-on: ubuntu-latest
    steps:
      - uses: anmol098/waka-readme-stats@master
        with:
          WAKATIME_API_KEY: ${{ secrets.WAKATIME_API_KEY }}
          GH_TOKEN: ${{ secrets.GH_TOKEN }}

**Reply by sharathsDev** [reply]
`https://github.com/anmol098/waka-readme-stats`

**Reply by Sahil9214** [reply]
bro can you tell the full step.We need to create a new repository or what we need to do?

**Reply by Lesuuh** [reply]
hey sahil9214, how you found the full steps, if yes, can you link me?

**Comment by SirToby-Gore** [comment]
Thanks man
